src/Security/Authentication/Negotiate/src/Internal/LdapAdapter.cs

```diff
@@ -109,18 +111,26 @@ private static void GetNestedGroups(LdapConnection connection, ClaimsIdentity pr
                 logger.LogWarning($"More than one response received for query: {filter} with distinguished name: {distinguishedName}");
             }
 
-            var group = searchResponse.Entries[0]; //Get the object that was found on ldap
-            string name = group.DistinguishedName;
-            retrievedClaims.Add(name);
```

Comment:
Usually, we use a group Common Name (CN, like `SomeGroup`) as a claim name. In this particular instance, a Distinguished Name (DN, like `CN=SomeGroup,DC=domain,DC=com`) was used instead, which I think is incorrect.
Possibly a typo, was introduced in #25075